src/transformers/tokenization_tapas.py

```diff
@@ -323,6 +326,829 @@ def save_vocabulary(self, save_directory: str, filename_prefix: Optional[str] =
+        verbose: bool = True,
+        prepend_batch_axis: bool = False,
+        **kwargs
+    ) -> BatchEncoding:
+
+        # Backward compatibility for 'truncation_strategy', 'pad_to_max_length'
+        padding_strategy, truncation_strategy, max_length, kwargs = self._get_padding_truncation_strategies(
+            padding=padding,
+            truncation=truncation,
+            max_length=max_length,
+            pad_to_multiple_of=pad_to_multiple_of,
+            verbose=verbose,
+            **kwargs,
+        )
+
+        encoded_inputs = {}
+
+        # This can be retrieved from the encoding step, which prevents recomputing.
+        # We still need to handle recomputing as `prepare_for_model` should be callable on raw IDs/table/query as well.
+        if (
+            "table_data" not in kwargs
+            or "query_tokens" not in kwargs
+            or (
+                ("table_data" in kwargs and kwargs["table_data"] is None)
+                and ("query_tokens" in kwargs and kwargs["query_tokens"] is None)
+            )
+        ):
+            table_tokens = self._tokenize_table(raw_table)
+            num_rows = self._get_num_rows(raw_table, self.drop_rows_to_fit)
+            num_columns = self._get_num_columns(raw_table)
+            _, _, num_tokens = self._get_table_boundaries(table_tokens)
+            table_data = list(self._get_table_values(table_tokens, num_columns, num_rows, num_tokens))
+            query_tokens = self.tokenize(raw_query)
+        else:
+            table_data = kwargs["table_data"]
+            query_tokens = kwargs["query_tokens"]
+
+        total_len = (
+            len(query_ids) + len(table_ids) + (self.num_special_tokens_to_add(pair=True) if add_special_tokens else 0)
+        )
+
+        overflowing_tokens = []
+        if truncation_strategy != TruncationStrategy.DO_NOT_TRUNCATE and max_length and total_len > max_length:
+            query_ids, table_ids, overflowing_tokens = self.truncate_sequences(
+                query_ids,
+                pair_ids=table_ids,
+                num_tokens_to_remove=total_len - max_length,
+                truncation_strategy=truncation_strategy,
+                stride=stride,
+            )
+
+        if return_overflowing_tokens:
+            encoded_inputs["overflowing_tokens"] = overflowing_tokens
+            encoded_inputs["num_truncated_tokens"] = total_len - max_length
+
+        if add_special_tokens:
+            input_ids = self.build_inputs_with_special_tokens(query_ids, table_ids)
+        else:
+            input_ids = query_ids + table_ids
+
+        encoded_inputs["input_ids"] = input_ids
+
+        segment_ids = self.create_segment_token_type_ids_from_sequences(query_ids, table_data)
+        column_ids = self.create_column_token_type_ids_from_sequences(query_ids, table_data)
+        row_ids = self.create_row_token_type_ids_from_sequences(query_ids, table_data)
+        prev_label_ids = [0] * len(row_ids)
```

Comment:
You're correct, this is my bad. I will fix this.

Reply:
Ok, maybe it's better to wait with this until the issue mentioned above is resolved. It seems to be that this can be implemented in a better way than in the original implementation.

Reply:
Update: after receiving a response from one of the authors (see issue above), I now understand how these are created in the original implementation (I actually implemented it in the wrong way with that dictionary). The correct way to do this (in case of a batch), is to set the `prev_label_ids` equal to `get_answer_ids(queries[index - 1])`, with index indicating whether the question is the first, second,... in a batch (note that a batch should always contain questions that refer to the same table). This function in turn calls `_get_answer_ids(column_ids, row_ids, question)`, with the column_ids and row_ids of the **current** table-question pair. So it's important that before calling get_answer_ids, the column_ids and row_ids are coming from to the current table-question pair.

This will require some changes, also to the signature of several methods (such as `get_answer_ids`, which is currently accepting a lot more parameters than just `question`), to reflect the original implementation.

Reply:
Done